app/main.py

```diff
@@ -1,3 +1,77 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        if self.km == 1:
+            return f"Distance: {self.km} kilometer."
+        else:
+            return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        elif isinstance(other, (float, int)):
+            return Distance(km=self.km + other)
+
+    def __iadd__(self, other: "Distance") -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        elif isinstance(other, (float, int)):
+            self.km += other
+        else:
+            raise TypeError("Operand must be a Distance instance or a number.")
+        return self
+
+    def __mul__(self, other: "Distance") -> "Distance":
+        return Distance(km=self.km * other)
+
+    def __truediv__(self, other: "Distance") -> "Distance":
+        if other == 0:
+            raise TypeError("zero division error")
+        else:
+            return Distance(km=round(self.km / other, 2))
```

Comment:
The `__truediv__` method should raise a `ZeroDivisionError` instead of a `TypeError` when dividing by zero. Additionally, the method should handle division with both `Distance` instances and numeric types (int, float). Update the type hint and logic to handle both cases.